clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -4377,8 +4384,12 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
                                                  ? llvm::Intrinsic::smax
                                                  : llvm::Intrinsic::umax,
                                              Op0, Op1, nullptr, "elt.max");
-    } else
-      Result = Builder.CreateMaxNum(Op0, Op1, /*FMFSource=*/nullptr, "elt.max");
+    } else {
+      FastMathFlags FMF;
+      FMF.setNoSignedZeros(true);
+      FMFSource FMFSrc(FMF);
+      Result = Builder.CreateMaxNum(Op0, Op1, /*FMFSource=*/FMFSrc, "elt.max");
```

Comment:
This is more verbose than necessary. FMFSource shouldn't need to appear, it's there to be an implicit conversion source.

I also thought the intent was you could do FastMathFlags::NoSignedZeros without the temporary structure

Reply:
I have some try to create FMF inline, while failed.